coreUtilities/src/main/java/org/opensearch/migrations/transformation/rules/IndexMappingTypeRemoval.java

```diff
@@ -0,0 +1,54 @@
+package org.opensearch.migrations.transformation.rules;
+
+import java.util.Map.Entry;
+
+import org.opensearch.migrations.transformation.CanApplyResult;
+import org.opensearch.migrations.transformation.CanApplyResult.Unsupported;
+import org.opensearch.migrations.transformation.TransformationRule;
+import org.opensearch.migrations.transformation.entity.Index;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Supports transformation of the Index Mapping types that were changed from mutliple types to a single type between ES 6 to 7
+ */
+public class IndexMappingTypeRemoval implements TransformationRule<Index> {
+
+    @Override
+    public CanApplyResult canApply(final Index index) {
+        final var mappingNode = index.raw().get("mappings");
+
+        if (mappingNode == null || mappingNode.isObject()) {
+            return CanApplyResult.NO;
+        }
+
+        // Detect multiple type mappings, eg:
+        // { mappings: [{ foo: {...}}, { bar: {...} }] } } or
+        // { mappings: [{ foo: {...}, bar: {...}] } }
+        if (mappingNode.size() > 1 || mappingNode.get(0).size() > 1) {
+            return new Unsupported("Multiple mapping types are not supported");
+        }
+
+        // There is a type under mappings, e.g. { mappings: [{ foobar: {...} }] }
+        return CanApplyResult.YES;
```

Comment:
Why isn't this just a boolean? (ignore the earlier comments on the class if the answer is that it should be)